src/views/TechnicalAssistance/RequestForm/Basic.tsx

```diff
@@ -1,34 +1,685 @@
+    UpdateTrbRequestFormVariables
+  >(UpdateTrbRequestFormQuery);
+
+  const initialValues = nullFillObject(request.form, basicBlankValues);
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isValid, isSubmitting, isDirty, dirtyFields }
+  } = useForm<FormFieldProps<TrbRequestFormBasic>>({
+    resolver: yupResolver(basicSchema),
+    defaultValues: {
+      name: request.name,
+      ...initialValues
+    }
+  });
+
+  // Using this instead of formState.isValid since it's not the same
+  const hasErrors = Object.keys(errors).length > 0;
+
+  // Scroll to the error summary when there are changes after submit
+  useEffect(() => {
+    if (hasErrors) {
+      const err = document.querySelector('.usa-alert--error');
+      err?.scrollIntoView();
+    }
+  }, [errors, hasErrors]);
+
+  const submit = useCallback<StepSubmit>(
+    callback =>
+      // Start the submit promise
+      handleSubmit(
+        // Validation passed
+        async formData => {
+          // Submit the input only if there are changes
+          if (isDirty) {
+            const { id } = request;
+            const { name } = formData;
+
+            // Only send changed fields for a partial update to the input object
+            const input: any = pick(formData, Object.keys(dirtyFields));
+
+            // Convert '' back to null for the backend
+            // so that cleared inputs on the client are actually removed
+            Object.entries(input).forEach(([key, value]) => {
+              if (value === '') input[key] = null;
+            });
+
+            // Handle the clearing out of toggled off fields.
+            // Unmounted fields are removed from the form data and do not get marked
+            // as dirty, so they need to be set to null to be cleared.
+            if (input.hasSolutionInMind === false) {
+              input.proposedSolution = null;
+            }
+            if (input.hasExpectedStartEndDates === false) {
+              input.expectedStartDate = null;
+              input.expectedEndDate = null;
+            }
+            if ('collabGroups' in input) {
+              Object.values(TRBCollabGroupOption).forEach(option => {
+                if (!input.collabGroups?.includes(option)) {
+                  input[`collabDate${upperFirst(camelCase(option))}`] = null;
+                  if (option === 'OTHER') {
+                    input.collabGroupOther = null;
+                  }
+                }
+              });
+            }
+
+            // Some object adjustments
+            const variables: any = {};
+
+            input.trbRequestId = id; // Use the id from the request object
+            variables.id = id;
+
+            // Move the name from the form fields object to changes
+            if ('name' in input) {
+              delete input.name;
+              variables.changes = { name };
+            }
+
+            variables.input = input;
+
+            await updateForm({ variables });
```

Comment:
I think this mutation could use some error handling, both in the form of an error from the request, and also a catch block for server errors.

Reply:
[Patch](https://github.com/CMSgov/easi-app/pull/1820/commits/0480bfd327a32df2b16dc05b2b18f406699cf203)